src/assets/src/components/preferences.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import { Alert, Button, Form } from "react-bootstrap";
+import PhoneInput from "react-phone-input-2";
+import 'react-phone-input-2/lib/style.css'
+
+import * as api from "../services/api";
+import { MyUser } from "../models";
+import { ErrorDisplay, FormError, checkForbiddenError, LoadingDisplay, LoginDialog, Breadcrumbs } from "./common";
+import { usePromise } from "../hooks/usePromise";
+import { redirectToLogin } from "../utils";
+import { PageProps } from "./page";
+
+const validatePhoneNumber = (phone: string, countryDialCode: string): Error[] => {
+    return [
+        (countryDialCode === '1' && phone.length !== 11)
+            && new Error("The phone number entered was invalid; USA phone numbers must have 11 digits."),
+    ].filter(e => e) as Error[];
+}
```

Comment:
My thinking here was that most validations can return multiple errors, and it's useful to be able to see them all at once rather than one at a time, although we're only performing a single validation so far.

Reply:
Good idea - moved it.